scripts/code_style.py

```diff
@@ -174,22 +182,23 @@ def main() -> int:
     parser.add_argument('-f', '--fix', action='store_true',
                         help=('modify source files to fix the code style '
                               '(default: print diff, do not modify files)'))
-    # --files is almost useless: it only matters if there are no files
-    # ('code_style.py' without arguments checks all files known to Git,
-    # 'code_style.py --files' does nothing). In particular,
-    # 'code_style.py --fix --files ...' is intended as a stable ("porcelain")
-    # way to restyle a possibly empty set of files.
-    parser.add_argument('--files', action='store_true',
-                        help='only check the specified files (default with non-option arguments)')
+    parser.add_argument('--subset', action='store_true',
+                        help=('check a subset of the files known to git '
+                              '(default: check all files passed as arguments, '
+                              'known to git or not)'))
     parser.add_argument('operands', nargs='*', metavar='FILE',
                         help='files to check (if none: check files that are known to git)')
 
     args = parser.parse_args()
 
-    if args.files or args.operands:
-        src_files = args.operands
-    else:
-        src_files = get_src_files()
+    all_src_files = get_src_files()
+    src_files = args.operands if args.operands else all_src_files
+    if args.subset:
+        # We are to check a subset of the default list
+        src_files = [f for f in args.operands if f in all_src_files]
+        skip_src_files = [f for f in args.operands if f not in src_files]
```

Comment:
We don't have enough files for it to really matter, but this is quadratic in the number of elements in `all_src_files`, and it would be very easy to make it quasi-linear.
```suggestion
        covered = frozenset(all_src_files)
        src_files = [f for f in args.operands if f in covered]
        skip_src_files = [f for f in args.operands if f not in covered]
```

(Out of curiosity, I timed it, and the quadratic version is about 500× slower, but it's only about a millisecond.)

Reply:
Right, the two versions don't have much difference, but the readability is better in second version. I also timed it, but it didn't show such a gap, only 2x slower at most. I think the first version would make sense only when there were plenty of skipped files in `arg.operands`, which would not be a common case.